app/controllers/payment/connection/currencies_controller.rb

```diff
@@ -6,16 +6,16 @@ class CurrenciesController < ApplicationController
 
       def show
         render json: {
-          supported_currencies: current_publisher.wallet_provider.supported_currencies,
-          default_currency: current_publisher.wallet_provider.default_currency,
+          supported_currencies: current_publisher.selected_wallet_provider.supported_currencies,
```

Comment:
Will this crash if the selected_wallet_provider is nil?

Reply:
Yes, but that's intended with the changes to the default currency workflow. 👍